glommio/src/executor/multitask.rs

```diff
@@ -185,10 +212,71 @@ impl LocalExecutor {
     ///
     /// Returns an option rapping the task.
     pub(crate) fn get_task(&self) -> Option<Runnable> {
-        self.local_queue.pop()
+        let mut run_next = self.run_next.borrow_mut();
+
+        let ret = match (
+            run_next.take(),
+            self.run_next_count.get() < MAX_UNFAIR_SCHEDULING,
+        ) {
+            (Some(runnable), true) => {
+                // run the next task immediately
+                self.run_next_count.replace(self.run_next_count.get() + 1);
+                Some(runnable)
+            }
+            (Some(runnable), false) => {
+                // pop from the queue instead
+                self.run_next_count.replace(0);
+                let ret = self.local_queue.pop();
+                self.local_queue.push(runnable);
+                ret
+            }
+            (_, _) => {
+                self.run_next_count.replace(0);
+                self.local_queue.pop()
+            }
+        };
+
+        if let Some(ref runnable) = ret {
+            self.last_popped.replace(Some(RunnableId::from(runnable)));
```

Comment:
Shouldn't we always set `last_popped`, even if just setting it to `None`?

Reply:
Good point!